source/runtime.cpp

```diff
@@ -243,9 +243,30 @@ void reshade::runtime::on_present()
 
 bool reshade::runtime::load_effect(const std::filesystem::path &path, size_t index)
 {
+	const std::string effect_name = path.filename().u8string();
+	const ini_file preset(_current_preset_path); // ini_file::load_cache is not thread-safe, so load from file here
+
 	effect &effect = _effects[index]; // Safe to access this multi-threaded, since this is the only call working on this effect
+	effect = {};
 	effect.source_file = path;
-	effect.compile_sucess = true;
+	effect.compiled = true;
+
+	if (!_effect_load_skipping || _load_option_disable_skipping)
+		effect.skipped = false;
+	else if (std::vector<std::string> techniques; !preset.get({}, "Techniques", techniques))
```

Comment:
I would consider doing a quick check to see if there is any `@` symbol in the technique list. If not, enable all effects, since the preset is using the old format. If yes, do the skip check.

Reply:
```cpp
else if (std::find_if(techniques.cbegin(), techniques.cend(), [&effect_name](const std::string &technique) {
		const size_t at_pos = technique.find('@') + 1;
		return at_pos == 0 || technique.find(effect_name, at_pos) == at_pos; }) != techniques.cend())
```

`at_pos` is overflow to 0 when `std::string::npos`. this is for no `@` symbol pattern.
so `skipped` is false.

Reply:
:+1: